crates/cext/src/transpiler/passes/commutative_cancellation.rs

```diff
@@ -0,0 +1,124 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::commutation_checker::get_standard_commutation_checker;
+use qiskit_transpiler::passes::cancel_commutations;
+use qiskit_transpiler::target::Target;
+
+/// @ingroup QkTranspilerPasses
+/// Run the CommutativeCancellation transpiler pass on a circuit.
+///
+/// This pass cancels the redundant (self-adjoint) gates through commutation relations.
+///
+/// @param circuit A pointer to the circuit to run CommutativeCancellation on. This circuit
+/// pointed to will be updated with the modified circuit if the pass is able to remove any gates.
+/// @param target The target for the CommutativeCancellation pass. The target is used to specify
```

Comment:
I don't fully understand this documentation -- the Python equivalent seems to say that the operation names in `target` specify what cancellations to consider, but I'm not reading the same here...

Reply:
The documentation wasn't clear on what it's used for to me. At least looking at: https://quantum.cloud.ibm.com/docs/en/api/qiskit/qiskit.transpiler.passes.CommutativeCancellation it didn't make any sense. So I looked at the code and tried to understand what it was doing there and describe it (specifically based on https://github.com/Qiskit/qiskit/blob/main/crates/transpiler/src/passes/commutation_cancellation.rs#L85-L101). I'm fine changing this to whatever you think is descriptive here.